presto-iceberg/src/main/java/com/facebook/presto/iceberg/TypeConverter.java

```diff
@@ -0,0 +1,438 @@
+import static java.util.stream.Collectors.toList;
+import static org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory.getCharTypeInfo;
+import static org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory.getListTypeInfo;
+import static org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory.getMapTypeInfo;
+import static org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory.getStructTypeInfo;
+import static org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory.getVarcharTypeInfo;
+
+public final class TypeConverter
+{
+    public static final String ORC_ICEBERG_ID_KEY = "iceberg.id";
+    public static final String ORC_ICEBERG_REQUIRED_KEY = "iceberg.required";
+    public static final String ICEBERG_LONG_TYPE = "iceberg.long-type";
+
+    private TypeConverter() {}
+
+    public static Type toPrestoType(org.apache.iceberg.types.Type type, TypeManager typeManager)
+    {
+        switch (type.typeId()) {
+            case BOOLEAN:
+                return BooleanType.BOOLEAN;
+            case BINARY:
+            case FIXED:
+                return VarbinaryType.VARBINARY;
+            case DATE:
+                return DateType.DATE;
+            case DECIMAL:
+                Types.DecimalType decimalType = (Types.DecimalType) type;
+                return DecimalType.createDecimalType(decimalType.precision(), decimalType.scale());
+            case DOUBLE:
+                return DoubleType.DOUBLE;
+            case LONG:
+                return BigintType.BIGINT;
+            case FLOAT:
+                return RealType.REAL;
+            case INTEGER:
+                return IntegerType.INTEGER;
+            case STRING:
+                return VarcharType.createUnboundedVarcharType();
+            case LIST:
+                Types.ListType listType = (Types.ListType) type;
+                return new ArrayType(toPrestoType(listType.elementType(), typeManager));
+            case MAP:
+                Types.MapType mapType = (Types.MapType) type;
+                TypeSignature keyType = toPrestoType(mapType.keyType(), typeManager).getTypeSignature();
+                TypeSignature valueType = toPrestoType(mapType.valueType(), typeManager).getTypeSignature();
+                return typeManager.getParameterizedType(StandardTypes.MAP, ImmutableList.of(TypeSignatureParameter.of(keyType), TypeSignatureParameter.of(valueType)));
+            case STRUCT:
+                List<Types.NestedField> fields = ((Types.StructType) type).fields();
+                return RowType.from(fields.stream()
+                        .map(field -> new RowType.Field(Optional.of(field.name()), toPrestoType(field.type(), typeManager)))
+                        .collect(toImmutableList()));
+            default:
+                throw new UnsupportedOperationException(format("Cannot convert from Iceberg type '%s' (%s) to Presto type", type, type.typeId()));
+        }
+    }
+
+    public static org.apache.iceberg.types.Type toIcebergType(Type type)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.